dtls/server-dtls13-event.c

```diff
@@ -0,0 +1,588 @@
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ *=============================================================================
+ *
+ * Single threaded example of a DTLS 1.3 server for instructional/learning
+ * purposes. This example can handle multiple simultaneous connections by using
+ * the libevent library to handle the event loop. Please note that this example
+ * is not thread safe as access to global objects is not protected.
+ */
+
+#include <wolfssl/options.h>
+#include <stdio.h>                  /* standard in/out procedures */
+#include <stdlib.h>                 /* defines system calls */
+#include <string.h>                 /* necessary for memset */
+#include <netdb.h>
+#include <sys/socket.h>             /* used for all socket calls */
+#include <netinet/in.h>             /* used for sockaddr_in */
+#include <arpa/inet.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/error-ssl.h>
+#include <errno.h>
+#include <signal.h>
+#include <unistd.h>
+
+/* Requires libevent */
+#include <event2/event.h>
+
+#include "dtls-common.h"
+
+#define SERV_PORT   11111           /* define our server port number */
+#define MSGLEN      4096
+#define QUICK_MULT  4               /* Our quick timeout multiplier */
+#define CHGOODCB_E  (-1000)         /* An error outside the range of wolfSSL
+                                     * errors */
+#define CONN_TIMEOUT 10             /* How long we wait for peer data before
+                                     * closing the connection */
+
+typedef struct conn_ctx {
+    struct conn_ctx* next;
+    WOLFSSL* ssl;
+    struct event* readEv;
+    struct event* writeEv;
+    char waitingOnData:1;
+} conn_ctx;
+
+WOLFSSL_CTX*  ctx = NULL;
+struct event_base* base = NULL;
+WOLFSSL*      pendingSSL = NULL;
+int           listenfd = INVALID_SOCKET;   /* Initialize our socket */
+conn_ctx* active = NULL;
+struct event* newConnEvent = NULL;
+
+static void sig_handler(const int sig);
+static void free_resources(void);
+static void newConn(evutil_socket_t fd, short events, void* arg);
+static void dataReady(evutil_socket_t fd, short events, void* arg);
+static int chGoodCb(WOLFSSL* ssl, void*);
+static int hsDoneCb(WOLFSSL* ssl, void*);
+static int newPendingSSL(void);
+static int newFD(void);
+static void conn_ctx_free(conn_ctx* connCtx);
+
+int main(int argc, char** argv)
+{
+    /* Loc short for "location" */
+    char          caCertLoc[] = "../certs/ca-cert.pem";
```

Comment:
You need `const` on these. Consider having the cert file paths in the shared header as macro.

Reply:
`const` added and moved to common header.